crypto/fipsmodule/ec/ec_key.c

```diff
@@ -332,49 +337,44 @@ int EC_KEY_check_key(const EC_KEY *eckey) {
   return 1;
 }
 
-static int EVP_EC_KEY_check_fips(EC_KEY *key) {
-  uint8_t msg[16] = {0};
-  size_t msg_len = 16;
-  int ret = 0;
-  uint8_t* sig_der = NULL;
-  EVP_PKEY *evp_pkey = EVP_PKEY_new();
-  EVP_MD_CTX ctx;
-  EVP_MD_CTX_init(&ctx);
-  const EVP_MD *hash = EVP_sha256();
-  size_t sign_len;
-  if (!evp_pkey ||
-      !EVP_PKEY_set1_EC_KEY(evp_pkey, key) ||
-      !EVP_DigestSignInit(&ctx, NULL, hash, NULL, evp_pkey) ||
-      !EVP_DigestSign(&ctx, NULL, &sign_len, msg, msg_len)) {
-    goto err;
+int EC_KEY_check_key(const EC_KEY *eckey) {
```

Comment:
does it make sense to simplify the code and leave only `EC_KEY_check_fips` now that the performance difference in FIPS/non_FIPS is not that big?